crates/testing/src/helpers.rs

```diff
@@ -128,6 +134,105 @@ pub async fn build_system_handle<
     .expect("Could not init hotshot")
 }
 
+/// create the [`SystemContextHandle`] from a launcher
+/// # Panics
+/// if cannot create a [`HotShotInitializer`]
+pub async fn build_system_handle_from_launcher<
```

Comment:
I'm hesitant to make such a specific change for the tests in this case, especially since you have them working. While it's not ideal to spin up the entire SystemContext for unit tests, this method unfortunately creates a rift in what is the accepted testing framework, and what is done here. If we made this change, we'd want to propagate it to *all* tests, lest we end up in the situation where specific knowledge of this specific suite rests solely with someone who is not on the team. 

I am not opposed to the outright removal of this in theory, though I think for this PR, we should stick to the existing spawner.

Reply:
@jparr721, I understand your concerns about consistency in our testing framework. To clarify, the existing `hotshot-testing::helpers::build_system_handle()` function remains unchanged, and all current test cases continue to work as expected.

The new API, `hotshot-testing::helpers::build_system_handle_from_launcher()`, was added to address issue #3160, which requires 'Permute the inputs and verify if the tests still pass.'. While `build_system_handle()` uses default test configuration parameters, the new function provides flexibility by allowing modifications to the setup.

I suggest we review its usage in the test scenarios. If it doesn’t add significant value, we can consider removing it. Specifically, 

- Validating vote collection and processing.
- Ensuring non-leader nodes correctly handle DA votes.

Reply:
Why not just use the `random!` macro to get permuted inputs in that case?

Reply:
Thank you, @jparr721, for your suggestion. I agree that utilizing random!() could be an effective way to introduce fuzziness in test configuration parameters. I will explore this approach and provide an update on the outcome.